wherehows-web/app/components/datasets/containers/dataset-owner-list.ts

```diff
@@ -0,0 +1,58 @@
+import Component from '@ember/component';
+import { get, set, computed } from '@ember/object';
+import ComputedProperty from '@ember/object/computed';
+import { IAvatar } from 'wherehows-web/typings/app/avatars';
+import { IOwner, IOwnerResponse } from 'wherehows-web/typings/api/datasets/owners';
+import { task } from 'ember-concurrency';
+import { readDatasetOwnersByUrn } from 'wherehows-web/utils/api/datasets/owners';
+import { arrayMap } from 'wherehows-web/utils/array';
+import { getAvatarProps } from 'wherehows-web/constants/avatars/avatars';
+
+export default class DatasetOwnerListContainer extends Component {
+  constructor() {
+    super(...arguments);
+
+    this.owners || (this.owners = []);
+  }
+
+  /**
+   * Urn for the related dataset
+   * @type {string}
+   * @memberof DatasetOwnerListContainer
+   */
+  urn: string;
+
+  /**
+   * The owners for the dataset
+   * @type {Array<IOwner>}
+   * @memberof DatasetOwnerListContainer
+   */
+  owners: Array<IOwner>;
+
+  /**
+   * Lists the avatar objects based off the dataset owners
+   * @type {ComputedProperty<Array<IAvatar>>}
+   * @memberof DatasetOwnerListContainer
+   */
+  avatars: ComputedProperty<Array<IAvatar>> = computed('owners', function(): Array<IAvatar> {
+    return arrayMap(getAvatarProps)(get(this, 'owners'));
+  });
+
+  didInsertElement() {
+    get(this, 'getOwnersTask').perform();
+  }
+
+  didUpdateAttrs() {
+    get(this, 'getOwnersTask').perform();
+  }
+
+  /**
+   * Reads the owners for this dataset
+   * @type {Task<Promise<Array<IOwnerResponse>>, () => TaskInstance<Promise<IOwnerResponse>>>}
+   */
+  getOwnersTask = task(function*(this: DatasetOwnerListContainer): IterableIterator<Promise<IOwnerResponse>> {
```

Comment:
Will defer to your knowledge on this, but based on the two render hooks above that call this, I am just a little worried there is potential to run this twice after first render.

Reply:
Why would that happen? `update` can only be invoked post render